src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -104,13 +104,6 @@ CHIP_ERROR NrfWiFiDriver::CommitConfiguration()
 
 CHIP_ERROR NrfWiFiDriver::RevertConfiguration()
 {
-    // Disconnection should happen automatically when WiFiManager::Connect() is called.
-    // Here we do it also explicitly to ping the Connectivity Manager which
-    // will send the disconnection event as a result.
-    // TODO: refactor when the callback-based supplicant API is incorporated
-    WiFiManager::Instance().DisconnectStation();
-    ConnectivityMgr().SetWiFiStationMode(ConnectivityManager::kWiFiStationMode_Disabled);
-
     LoadFromStorage();
 
     if (mStagingNetwork.IsConfigured())
```

Comment:
Should we disconnect if not configured?

Reply:
I think we have 2 scenarios here:
- not configured and not connected (because the previous connect failed, e.g. due to invalid credentials)
- not configured, connected and the DNS-SD failed for whatever reason

For the first case we are getting mentioned bus fail, I guess it is just redundant.
I am not sure how to filter the second scenario though - in this case we should disconnect for sure.

Reply:
As discussed offline, JIRA ticket (KRKNWK-14975) created for a scenario when the connection is established but the service discovery fails.